app/server/src/main/scala/org/bitcoins/server/BitcoindRpcBackendUtil.scala

```diff
@@ -264,16 +299,21 @@ object BitcoindRpcBackendUtil extends Logging {
       interval: FiniteDuration = 10.seconds)(implicit
       system: ActorSystem,
       ec: ExecutionContext): Future[Cancellable] = {
-    bitcoind.getBlockCount.map { startCount =>
+    val walletSyncStateF = wallet.getSyncState()
+    val resultF: Future[Cancellable] = for {
+      walletSyncState <- walletSyncStateF
+    } yield {
       val numParallelism = Runtime.getRuntime.availableProcessors()
-      val atomicPrevCount: AtomicInteger = new AtomicInteger(startCount)
+      val atomicPrevCount: AtomicInteger = new AtomicInteger(
+        walletSyncState.height)
```

Comment:
There is a slight change here, we no longer are comparing bitcoind heights, we are comparing wallet height and bitcoind height. Is this a significant change?

Reply:
this shouldn't be a big change, this is probably a small bug fix. before we would call `startBitcoindBlockPolling` we would make sure we were synced to the api. this should fix an edge case where a block is mined between the syncing being done and the block polling being started.